### PR TITLE
LY-LA119.solar

### PR DESCRIPTION
**Pull request by fengly20**
#839 

Passed tests.

### CODE REVIEW

File: R/zchunk_L223.electricity.R [hunk 1]
```diff
@@ -61,7 +61,7 @@ module_energy_L223.electricity <- function(command, ...) {
              FILE = "energy/A23.globaltech_co2capture",
              "L114.RsrcCurves_EJ_R_wind",
              "L118.out_EJ_R_elec_hydro_Yfut",
-             FILE = "temp-data-inject/L119.Irradiance_rel_R",
+             "L119.Irradiance_rel_R",
```
Comment (bpbond) [inline R/zchunk_L223.electricity.R]: Thanks for doing this!

File: R/zchunk_LA119.solar.R [hunk 2]
```diff
@@ -27,44 +27,114 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
+      irradiance.area.R <- irradiance_avg <- irradiance_avg_rel <- iso <-
+      region_GCAM3 <- value <- year <- NULL
+
     # ===================================================
+    # 2. Perform computations
+    # The irradiance data contained other * region categories which we will have to downscale
+    # Seperate the other * region from the ones with valid country codes that do not need additonal
+    # downscaling
+    Smith_irradiance_ctry_kwh %>%
+      filter(iso %in% grep("OTHER", iso, value = T)) %>%
```
Comment (bpbond) [inline R/zchunk_LA119.solar.R]: `TRUE`

File: R/zchunk_LA119.solar.R [hunk 3]
```diff
@@ -27,44 +27,114 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      inner_join(L119.Other_ctry, "iso") ->
+      L119.LC_bm2_other_ctry # intermediate tibble
+
+    L119.LC_bm2_other_ctry %>%
+      group_by(region_GCAM3) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      rename(Area.bm2.R = value) %>%
```
Comment (bpbond) [inline R/zchunk_LA119.solar.R]: Why not `Area.bm2.R = sum(value)` above and remove this line?

File: R/zchunk_LA119.solar.R [hunk 4]
```diff
@@ -27,44 +27,114 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
+      mutate(Area.share = value / Area.bm2.R) %>%
+      # Now include the irradiance data at the regional level and multiply the share to downscale to the
+      # country
+      inner_join(L119.Irradiance_kwh_otherR, by = c( "region_GCAM3", "iso", "country_name", "GCAM_region_ID")) %>%
+      mutate( irradiance = irradiance.R * Area.share,
+              irradiance.area = irradiance.area.R * Area.share,
+              dni = dni.R * Area.share,
+              dni.area = dni.area.R * Area.share) %>%
+      # Add the downscaled coutries in with the rest of the country irradiance data
+      select(one_of(names(L119.Irradiance_kwh_ctry))) %>%
```
Comment (bpbond) [inline R/zchunk_LA119.solar.R]: Just `select(names(...))` - we've upgraded to dplyr 0.7 exclusively now.

File: R/zchunk_LA119.solar.R [hunk 5]
```diff
@@ -27,44 +27,114 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
+    L119.Irradiance_kwh_ctry %>%
+      filter(iso == "usa") %>%
+      mutate(irradiance_avg = irradiance / irradiance.area,
+             dni_avg = dni / dni.area) ->
+      L119.Irradiance_kwh_usa # intermediate tibble
+
+    L119.Irradiance_kwh_R %>%
+      mutate(irradiance_avg_rel = irradiance_avg / L119.Irradiance_kwh_usa[["irradiance_avg"]],
+             dni_avg_rel = dni_avg / L119.Irradiance_kwh_usa[["dni_avg"]]) %>%
+      mutate( irradiance_avg_rel = if_else(is.na(irradiance_avg_rel), 0.001, irradiance_avg_rel),
```
Comment (bpbond) [inline R/zchunk_LA119.solar.R]: Combine into a single `mutate` call

File: R/zchunk_LA119.solar.R [hunk 6]
```diff
@@ -27,44 +27,114 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+    L119.Irradiance_rel_R %>%
+      add_title("Relative solar irradiance by GCAM region") %>%
+      add_units("Unitless") %>%
+      add_comments("The irradiance and dni for aggregated 'OTHER_*' regions are first downscaled to country level then combined with irradiance and dni for rest countries to aggregate to GCAM regions. ") %>%
```
Comment (bpbond) [inline R/zchunk_LA119.solar.R]: Nice comments!

**Comment by bpbond** [comment]
Holding on @skim301 review.

File: R/zchunk_LA119.solar.R [hunk 7]
```diff
@@ -27,44 +27,113 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
...
+    # The irradiance data contained other * region categories which we will have to downscale
+    # Seperate the other * region from the ones with valid country codes that do not need additonal
+    # downscaling
+    Smith_irradiance_ctry_kwh %>%
+      filter(iso %in% grep("OTHER", iso, value = TRUE)) %>%
+      # Rename columns to indicate these are the regional values which will be used later
+      # to downscale to the country
+      mutate(region_GCAM3 = sub("OTHER_", "", iso)) %>%
+      select(-iso) %>%
+      rename(irradiance.R = irradiance,
```
Comment (skim301) [inline R/zchunk_LA119.solar.R]: I'm concerned that the ".R" used throughout may be confused with the R coding language. Could we use "reg" instead?

File: R/zchunk_LA119.solar.R [hunk 8]
```diff
@@ -1,23 +1,23 @@
 #' module_energy_LA119.solar
 #'
-#' Briefly describe what this chunk does.
+#' Computes relative average irradiance and direct normal irradiance by GCAM region for solar sector.
```
Comment (skim301) [inline R/zchunk_LA119.solar.R]: could you add "(dni)" so that the acronym is clear.

File: R/zchunk_LA119.solar.R (hunk 7 above)
Comment (fengly20) [reply R/zchunk_LA119.solar.R]: Good point! It confused me when I started re-writing the chunk. Using `reg` sounds good to me unless `.R` is also used in other chunks that we want to be consistent with. Do we have a decision? @bpbond

Comment (skim301) [reply R/zchunk_LA119.solar.R]: I think for file names, we need to leave alone. But for local variable names, it should be fine.

Comment (bpbond) [reply R/zchunk_LA119.solar.R]: Agreed--good suggestion @skim301 , thanks. No, I've never seen another ".R" variable in the data system.

**Comment by bpbond** [comment]
Thanks @fengly20 ! Nice job.

**Pull request merged**